### ISSUE TITLE
--indent 0 results in 4 spaces instead

### ISSUE THREAD

**Issue by avorima** [mikefarah/yq#1668]
**Describe the bug**

The using --indent with anything less than or equal to zero results in 4 spaces being used instead.

Version of yq: 4.33.3
Operating system: linux
Installed via: [pacman](https://archlinux.org/packages/community/x86_64/go-yq/), but also reproducible with locally built binary

**Input Yaml**

data1.yml:
```yaml
list:
- foo
```

**Command**

```
yq --indent 0 -i '.list[0] = "bar"' data1.yaml
```

**Actual behavior**

```yaml
list:
    - bar
```

**Expected behavior**

```yaml
list:
- bar
```

**Additional context**

This isn't limited to the `-i` (inplace) flag.

**Comment by mikefarah** [comment]
Yeah this is what the underlying yaml library does - I believe there is already a bug raised against it - but that library isn't getting much love atm.

I plan on switching to another yaml library hopefully sometime this year. It will be a reasonably big job though :/

**Comment by avorima** [comment]
Oh, I just found your issue https://github.com/mikefarah/yq/issues/825. What library would fix this?

**Comment by PatMyron** [comment]
Similar to https://github.com/jqlang/jq/issues/1465, `--indent 0` also removes JSON newlines in `yq 4.45.1`
